### ISSUE TITLE
Deal gracefully with network timeout exceptions

### ISSUE THREAD

**Issue by siddharthist** [matterhorn-chat/matterhorn#492]
Matterhorn just sent me this message:
```
An unexpected error has occurred! The exception encountered was:
HostCannotConnect "mattermost.galois.com" [Network.Socket.connect: <socket: 47>: timeout (Connection
timed out)]
Please report this error at https://github.com/matterhorn-chat/matterhorn/issues
```
Since this is a rather pedestrian exception (and definitely not a problem with Matterhorn), it should probably not cause users to be asked to file an issue. 

## system details

```
matterhorn 50200.1.1
 using mattermost-api 50200.1.0
```
```
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.107, NixOS, 18.09.2394.2d656a97297 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(siddharthist): `""`
 - channels(root): `"nixos-18.09.2394.2d656a97297"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by jtdaugherty** [comment]
Thanks for the report!

**Comment by jtdaugherty** [comment]
This is now fixed in `develop`. Thanks!

**Comment by siddharthist** [comment]
Awesome!

**Comment by jtdaugherty** [comment]
This is now released in 50200.3.0.
